### PR TITLE
ci: Run checklist job as a separate workflow (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**

### CODE REVIEW

File: .github/workflows/checklist.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: PR Checklist
+
+on:
+  pull_request_target:
```
Comment (netroy) [inline .github/workflows/checklist.yml:4]: I'm hoping this change fixes the issue with community PRs

Comment (mutdmour) [reply .github/workflows/checklist.yml:4]: what's the issue?

**Review by mutdmour — APPROVED** [review]

File: .github/workflows/checklist.yml (hunk 1 above)
Comment (netroy) [reply .github/workflows/checklist.yml:4]: This jobs keeps failing for community PR. [example](https://github.com/n8n-io/n8n/actions/runs/5530465399).

**Comment by janober** [comment]
Got released with [`n8n@1.1.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.1.0)

**Pull request merged**